libs/libcommon/include/common/DateLUTImpl.h

```diff
@@ -284,6 +284,15 @@ class DateLUTImpl
     inline time_t toStartOfFiveMinute(time_t t) const { return t / 300 * 300; }
     inline time_t toStartOfFifteenMinutes(time_t t) const { return t / 900 * 900; }
 
+    inline time_t toStartOfTenMinutes(time_t t) const
+    {
+        if (offset_is_whole_number_of_hours_everytime)
```

Comment:
It's guaranteed that the offset is divisable by 15 minutes for all time zones since some reasonable year.
You can avoid this check.

Reply:
The implementation looks inconsistent with two functions above.

Reply:
Fixed in https://github.com/yandex/ClickHouse/pull/4298